test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Builder/LinkGenerationHelpersTest.cs

```diff
@@ -93,6 +94,60 @@ public void GenerateNavigationLink_WorksToGenerateExpectedNavigationLink_ForSing
             Assert.Equal(expectedNavigationLink, uri.AbsoluteUri);
         }
 
+        private ResourceContext GetOrderItemResourceForNewSingletonContainer()
+        {
+            // Arrange
+            IEdmSingleton myVipOrder = _model.Model.FindDeclaredSingleton("VipOrder");
+            IEdmEntityType vipOrderType = (IEdmEntityType)myVipOrder.Type;
+            IEdmNavigationProperty orderItemsProperty = vipOrderType.NavigationProperties().Single(x => x.ContainsTarget && x.Name == "OrderItems");
+            IEdmContainedEntitySet orderItems = (IEdmContainedEntitySet)myVipOrder.FindNavigationTarget(orderItemsProperty);
+            IEdmEntityType orderItem = _model.OrderItem;
+            IEdmNavigationProperty orderItemDetailsNav = orderItem.NavigationProperties().First();
+
+            var request = RequestFactory.CreateFromModel(_model.Model);
+
+            ODataPath path = new ODataPath(
+                    new SingletonSegment(myVipOrder),
+                    new NavigationPropertySegment(orderItemsProperty, orderItems));
+
+            ODataSerializerContext orderItemSerializerContext = ODataSerializerContextFactory.Create(_model.Model, orderItems, path, request);
+            orderItemSerializerContext.EdmProperty = orderItemDetailsNav;
+            ResourceContext orderItemResource = new ResourceContext(orderItemSerializerContext, orderItem.AsReference(), new { ID = 21 });
+            orderItemSerializerContext.ExpandedResource = orderItemResource;
+
+            return orderItemResource;
+        }
+
+        [Fact]
+        public void GenerateBaseODataPathSegments_WorksToGenerateExpectedPath_ForSingletonContainer()
+        {
+            // Arrange
+            ResourceContext orderItemResource = GetOrderItemResourceForNewSingletonContainer();
+
+            // Act
+            IList<ODataPathSegment> newPaths = orderItemResource.GenerateBaseODataPathSegments();
+
+            // Assert
+            Assert.Equal(3, newPaths.Count);
+            Assert.IsType<Microsoft.OData.UriParser.SingletonSegment>(newPaths[0]); // VipOrder
+            Assert.IsType<Microsoft.OData.UriParser.NavigationPropertySegment>(newPaths[1]); // OrderItems
+            Assert.IsType<Microsoft.OData.UriParser.KeySegment>(newPaths[2]); // 21
```

Comment:
21 ?

Reply:
21 is the expected ID for this path segment